### PR TITLE
White Cane for Blind People

### PR DESCRIPTION
**Pull request by forkeyboards**
## About the PR

This PR adds a white cane, which spawns on blind characters when they join the game. I added a video showing the description and all angles of the cane. The cane does 5 damage, and 8 damage when wielded. It also shows the cane spawning on a blind player.

Credit to Finket for modifying the .cs files, and I did the spriting and yamling.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Tristan Thomas
- add: Added a white cane, which spawns on blind players.

https://github.com/space-wizards/space-station-14/assets/91704530/bcbfa854-110f-4fd9-ad58-dbfbca9b4c5f

### CODE REVIEW

File: Content.Server/Traits/TraitSystem.cs [hunk 1]
```diff
@@ -41,6 +44,18 @@ private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent args)
                 comp.Owner = args.Mob;
                 EntityManager.AddComponent(args.Mob, comp);
             }
+
+            // Add item required by the trait
+            if (traitPrototype.TraitGear != null)
+            {
+                if (!TryComp(args.Mob, out HandsComponent? handsComponent))
+                    return;
```
Comment (metalgearsloth) [inline Content.Server/Traits/TraitSystem.cs:52]:
```suggestion
                if (!TryComp(args.Mob, out HandsComponent? handsComponent))
                    continue;
```

File: Content.Server/Traits/TraitSystem.cs [hunk 2]
```diff
@@ -41,6 +44,18 @@ private void OnPlayerSpawnComplete(PlayerSpawnCompleteEvent args)
...
+
+                var coords = EntityManager.GetComponent<TransformComponent>(args.Mob).Coordinates;
```
Comment (metalgearsloth) [inline Content.Server/Traits/TraitSystem.cs:54]:
```suggestion
                var coords = Transform(args.Mob).Coordinates;
```

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/white_cane.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Melee/white_cane.rsi
+    state: icon
+  - type: Item
+    size: 15
+    sprite: Objects/Weapons/Melee/white_cane.rsi
+  - type: MeleeWeapon
+    bluntStaminaDamageFactor: 1.60 
+    range: 1.85
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/white_cane.yml:15]: WHy the overwrites?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/white_cane.yml (hunk 3 above)
Comment (forkeyboards) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/white_cane.yml]: No particular reason. I have no real coding experience and didn't realize that this was unnecessary. I removed the lines

File: Content.Server/Traits/TraitSystem.cs (hunk 1 above)
Comment (forkeyboards) [reply Content.Server/Traits/TraitSystem.cs]: Changed, thanks for the feedback.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**